app/screens/home/search/results/file_options/file_options.tsx

```diff
@@ -0,0 +1,217 @@
+import {useTheme} from '@context/theme';
+import CopyPublicLink from '@screens/gallery/footer/copy_public_link';
+import DownloadWithAction from '@screens/gallery/footer/download_with_action';
+import {getFormattedFileSize, isImage, isVideo} from '@utils/file';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+const format = 'MMM DD YYYY HH:MM A';
+
+export const TOAST_MARGIN_BOTTOM = 40;
+
+const getStyleSheet = makeStyleSheetFromTheme((theme) => {
+    return {
+        container: {
+            marginTop: -20,
+        },
+        headerContainer: {
+            float: 'left',
+            marginTop: 20,
+            marginBottom: 8,
+        },
+        fileIconContainer: {
+            marginLeft: -10,
+            marginBottom: 8,
+            alignSelf: 'flex-start',
+        },
+        nameText: {
+            color: theme.centerChannelColor,
+            ...typography('Heading', 400, 'SemiBold'),
+        },
+        imageVideo: {
+            marginLeft: 10,
+            height: 72,
+            width: 72,
+        },
+        optionContainer: {
+            marginHorizontal: -20,
+            paddingHorizontal: 20,
+        },
+        infoContainer: {
+            marginVertical: 8,
+            alignItems: 'center',
+            flexDirection: 'row',
+        },
+        infoText: {
+            flexDirection: 'row',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        date: {
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            ...typography('Body', 200, 'Regular'),
+        },
+        toast: {
+            marginTop: 100,
+            alignItems: 'center',
+        },
+    };
+});
+
+type Props = {
+    canDownloadFiles: boolean;
+    enablePublicLink: boolean;
+    fileInfo: FileInfo;
+}
+const FileOptions = ({fileInfo, canDownloadFiles, enablePublicLink}: Props) => {
+    const theme = useTheme();
+    const style = getStyleSheet(theme);
+    const intl = useIntl();
+    const ref = useRef<any>();
+    const serverUrl = useServerUrl();
+    const [action, setAction] = useState<GalleryAction>('none');
+
+    const galleryItem = {...fileInfo, type: 'image'} as GalleryItemType;
+    const size = getFormattedFileSize(fileInfo.size);
+
+    const handleDownload = useCallback(() => {
+        setAction('downloading');
+    }, []);
+
+    const handleCopyLink = useCallback(() => {
+        setAction('copying');
+    }, []);
+
+    const handlePermalink = useCallback(async () => {
+        const fetchedPost = await fetchPostById(serverUrl, fileInfo.post_id, true);
+        const post = fetchedPost.post;
+        const rootId = post?.root_id || post?.id;
+        if (rootId) {
+            showPermalink(serverUrl, '', post.id, intl);
+        } else {
+            // what to do?
+        }
+    }, [fileInfo]);
+
+    const renderIcon = () => {
```

Comment:
@larkox I didn't think this needed a useCallback because 1) it is always being called, 2) It only depends on fileInfo which will not change, and it only returns a component.  

`renderHeader` depends oni it, but it only depends on fileInfo, which will not change.  Does that sound correct?

Reply:
LGTM